### PR TITLE
Feature: Handle FragmentsCompactedEvent and FragmentDeletedEvent

### PR DESCRIPTION
**Pull request by WLefever-Cegeka**
In this PR:
* Handling of FragmentsCompactedEvent (DeleteTime is set on Fragments)
* Deleting of Fragments whose DeleteTime has expired
* Handling of FragmentDeletedEvent

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -363,6 +372,17 @@ Templates for configuring the DCAT metadata can be found [here](templates/dcat)
 A detailed explanation on how to manage and retrieve the DCAT metadata can be found [here](ldes-server-admin/README.md#dcat-endpoints)
 or on the [swagger endpoint](#example-swagger-configuration) if it is configured.
 
+##### Example Compaction
+
+Compaction is a process that allows to merge immutable fragments that are underutilized (i.e. there are fewer members in the fragment than indicated in the `pageSize` of the view).
```
Comment (Yalz) [inline README.md:377]: That allows the server to merge ..

File: ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/eventhandler/FragmentDeleteTimeSetter.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.LdesFragmentIdentifier;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.repository.FragmentRepository;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+@Component
+public class FragmentDeleteTimeSetter {
```
Comment (Yalz) [inline ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/eventhandler/FragmentDeleteTimeSetter.java:14]: Most likely just poreferance, but FragmentDeletionTimeSetter seems a bit more fluent

File: ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/FragmentDeleterScheduler.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.events.fragmentation.FragmentDeletedEvent;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.entities.Fragment;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.repository.FragmentRepository;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FragmentDeleterScheduler {
```
Comment (Yalz) [inline ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/FragmentDeleterScheduler.java:11]: Same, personal preference but FragmentDeletionScheduler would be more fluent

File: ldes-fragmentisers/ldes-fragmentisers-common/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/FragmentDeleterSchedulerTest.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.context.ApplicationEventPublisher;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class FragmentDeleterSchedulerTest {
```
Comment (Yalz) [inline ldes-fragmentisers/ldes-fragmentisers-common/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/FragmentDeleterSchedulerTest.java:21]: Pref: FragmentDeletionScheduler

File: ldes-fragmentisers/ldes-fragmentisers-common/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/eventhandler/FragmentDeleteTimeSetterTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class FragmentDeleteTimeSetterTest {
```
Comment (Yalz) [inline ldes-fragmentisers/ldes-fragmentisers-common/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/eventhandler/FragmentDeleteTimeSetterTest.java:23]: pref: FragmentDeletionTimeSetterTest

**Review by Yalz — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (WLefever-Cegeka) [reply README.md:377]: updated

File: ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/eventhandler/FragmentDeleteTimeSetter.java (hunk 2 above)
Comment (WLefever-Cegeka) [reply ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/eventhandler/FragmentDeleteTimeSetter.java:14]: updated

File: ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/FragmentDeleterScheduler.java (hunk 3 above)
Comment (WLefever-Cegeka) [reply ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/FragmentDeleterScheduler.java:11]: updated

File: ldes-fragmentisers/ldes-fragmentisers-common/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/eventhandler/FragmentDeleteTimeSetterTest.java (hunk 5 above)
Comment (WLefever-Cegeka) [reply ldes-fragmentisers/ldes-fragmentisers-common/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/eventhandler/FragmentDeleteTimeSetterTest.java:23]: updated

**Pull request merged**